docs/demo/scripts/customise_interactions.py

```diff
@@ -76,45 +103,58 @@
 # Remember that a `push_check` requires sending a VQIP (a dictionary with a key for 'volume' and each pollutant
 # simulated).
 # %%
-reply = my_arc.send_push_check({"volume": 10, "phosphate": 1, "temperature": 10})
+reply = my_model.arcs["my_arc"].send_push_check(
+    {"volume": 10, "phosphate": 1, "temperature": 10}
+)
 print(reply)
 
 
 # %% [markdown]
 # The `Distribution` node replied that it can accept 0 water.
 #
 # But we might try and customise the `my_dist` object so that it calls some function before carrying on with its default push check.
-
-
+#
+# Let's have a look at what code we need to store in our extension module, which we have saved in a file called `custom_distribution_handler.py`.
+#
+# <details>
+# <summary>Source code</summary>
+# <pre><code class="language-python">
+# from wsimod.extensions import register_node_patch
+#
+# @register_node_patch("my_dist", "push_check_handler", item="Node")
+# def custom_handler_function(self, vqip, *args, **kwargs):
+#     """A custom handler function."""
+#     print("I reached a custom handler")
+#     return self.push_check_handler\["default"\](vqip)
+# </code></pre>
+#
+# </details>
```

Comment:
This does not appear correcly formatted when building the documentation. The tabs are all missing. 

<img width="678" alt="image" src="https://github.com/user-attachments/assets/c66b52f5-563d-450a-a886-0582c1299c80">

And if you just make this a code cell instead of embedding it into the markdown cell?

Reply:
well then it is registered twice and throws an error. I could of course register them on the fly but I was hoping to have them in files to create something as close to the use case as possible.

I guess I could say - you put these in a separate folder and then put something along the lines of `my_model.extensions = [os.path.join(scripts_folder, "custom_distribution_handler.py")]` in the markdown which at least doesn't have any significant indents or formatting?

Reply:
(just tagging you on this btw for visibility since I think need to decide on it before addressing other things @dalonsoa )

Reply:
This is super annoying. So, the usual ` ``` ` blocks don't work, right?

An option - which might be a good thing, to be honest - is to allow for a patch to be registered twice with just a warning, rather than raising an exception as it is now:

https://github.com/ImperialCollegeLondon/wsi/blob/5f879f3bdb9b3e00a4797714f86b418a216f144a/wsimod/extensions.py#L84

Then you mention why that warning. Or we can add a flag to the decorator, `override`, that defaults to `false`. So normally you will have an error if you try to register the same thing twice. But in this example you set is as `true` so there's no error. And, again, you explain the flag.

Reply:
The ``` do work, but they look completely different from the jupyter codeblocks. I think keeping them in codeblocks as actual code and warning for multiple patch targets is probably the sensible solution (especially since it is often biting me on debugging ;) )